geode-benchmarks/src/main/java/org/apache/geode/benchmark/tests/PartitionedWithDeltaAndUniqueObjectReferenceBenchmark.java

```diff
@@ -0,0 +1,242 @@
+import java.util.concurrent.atomic.AtomicReference;
+
+import benchmark.geode.data.Session;
+import org.junit.jupiter.api.Test;
+import org.yardstickframework.BenchmarkConfiguration;
+import org.yardstickframework.BenchmarkDriverAdapter;
+
+import org.apache.geode.benchmark.tasks.CreatePartitionedRegion;
+import org.apache.geode.benchmark.topology.ClientServerTopology;
+import org.apache.geode.cache.Region;
+import org.apache.geode.cache.client.ClientCache;
+import org.apache.geode.cache.client.ClientCacheFactory;
+import org.apache.geode.cache.client.ClientRegionShortcut;
+import org.apache.geode.cache.client.PoolFactory;
+import org.apache.geode.cache.client.PoolManager;
+import org.apache.geode.perftest.PerformanceTest;
+import org.apache.geode.perftest.Task;
+import org.apache.geode.perftest.TestConfig;
+import org.apache.geode.perftest.TestContext;
+import org.apache.geode.perftest.TestRunners;
+
+public class PartitionedWithDeltaAndUniqueObjectReferenceBenchmark implements PerformanceTest {
+
+  private static final String SUBSCRIPTION_POOL = "subscriptionPool";
+  private static final String ID = UUID.randomUUID().toString();
+
+  public PartitionedWithDeltaAndUniqueObjectReferenceBenchmark() {}
+
+  @Test
+  public void run() throws Exception {
+    TestRunners.defaultRunner().runTest(this);
+  }
+
+  @Override
+  public TestConfig configure() {
+    TestConfig config = GeodeBenchmark.createConfig();
+    config.threads(180);
+    config.durationSeconds(300);
+    config.warmupSeconds(30);
+    ClientServerTopology.configure(config);
+    config.before(new CreatePartitionedRegion(), SERVER);
+    config.before(new CreateClientPool(), CLIENT);
+    config.before(new CreateClientProxyRegionWithPool(), CLIENT);
+    config.workload(new PutWithDeltaTask(ID), CLIENT);
+    return config;
+  }
+
+  public static class PutWithDeltaTask extends BenchmarkDriverAdapter implements Serializable {
+
+    private Region<String, Session> region;
+    private final AtomicReference<String> sessionId = new AtomicReference<>(null);
+
+    private final String id;
+
+    private final List<String> existingSessionAttributeNames =
+        new ArrayList<>();
+
+    private final Random random = new Random(System.currentTimeMillis());
+
+    private Collection<Callable<Integer>> workloadTasks;
+
+    public PutWithDeltaTask(String id) {
+      this.id = id;
+    }
+
+    @Override
+    public void setUp(BenchmarkConfiguration cfg) throws Exception {
+      super.setUp(cfg);
+      this.workloadTasks = newSessionWorkloadTasks();
+      ClientCache cache = ClientCacheFactory.getAnyInstance();
+      cache.setCopyOnRead(true);
+      region = cache.getRegion("region");
+      save(Session.create(id));
+      this.sessionId.set(id);
+    }
+
+    @Override
+    public boolean test(Map<Object, Object> ctx) throws Exception {
+      int sessionAttributeCount = runSessionWorkload();
+
+      assert sessionAttributeCount == this.existingSessionAttributeNames.size();
```

Comment:
Leave `assert` out of here. We never run with asserts enabled. Write a unit test for this benchmark that asserts the behavior you expect.